rustc_codegen_spirv/src/symbols.rs

```diff
@@ -279,15 +414,169 @@ pub fn parse_attrs(
+                                                        arg.span(),
+                                                        &format!(
+                                                            "#[spirv({}(..))] attribute argument must be single identifier", 
+                                                            name.name.to_ident_string()
+                                                        ),
+                                                    );
+                                                }
+                                                    
+                                            });
+                                    }
+                                    match e.execution_model {
+                                        Fragment => {
+                                            //FIXME: Is this the correct default?
+                                            let origin_mode = origin_mode.unwrap_or(OriginUpperLeft);
+                                            e.execution_modes.push((origin_mode, ExecutionModeExtra::new([])));
+                                        },
+                                        GLCompute => {
+                                            let local_size = local_size.unwrap_or([1, 1, 1]);
+                                            e.execution_modes.push((LocalSize, ExecutionModeExtra::new(local_size)));
+                                        },
+                                        Kernel => {
+                                            if let Some(local_size) = local_size {
+                                                e.execution_modes.push((LocalSize, ExecutionModeExtra::new(local_size)));
+                                            }
+                                            if let Some(local_size_hint) = local_size_hint {
+                                                e.execution_modes.push((LocalSizeHint, ExecutionModeExtra::new(local_size_hint)));
+                                            }
+                                            // Reserved
+                                            /*if let Some(max_workgroup_size_intel) = max_workgroup_size_intel {
+                                                e.execution_modes.push((MaxWorkgroupSizeINTEL, ExecutionModeExtra::new(max_workgroup_size_intel)));
+                                            }*/
+                                        },
+                                        //TODO: Cover more defaults
+                                        _ => {},
+                                    }
+                                }
+                                _ => {}
+                            }
+                            a
+                        })
+                };
+                if result.is_none() {
+                    cx.tcx
+                        .sess
+                        .span_err(name.span, "unknown argument to spirv attribute");
                 }
+                result
```

Comment:
This whole thing is super nested, it'd be really nice if we could split it into some smaller functions.